src/UI/Actions.hs

```diff
@@ -860,7 +865,14 @@ abort = Action ["cancel"] (reset (Proxy :: Proxy v) (Proxy :: Proxy a))
 focus :: forall a v. (HasViewName v, HasName a, Focusable v a) => Action v a AppState
 focus = Action
   ["switch mode to " <> T.pack (show (name (Proxy :: Proxy a)))]
-  (switchFocus (Proxy :: Proxy v) (Proxy :: Proxy a))
+  (\s -> do
+    liftIO . view (asConfig . confLogSink) s . T.pack $
+      "switchFocus "
+        <> show (viewname (Proxy :: Proxy v)) <> " "
+        <> show (name (Proxy :: Proxy a))
```

Comment:
Type applications? `Proxy @a`?

Reply:
I thought about it.  I'm not against it but I didn't want to either (a) refactor the classes in this commit or (b) apply the functions to `undefined` as the (still currently required) argument.  Feel free to open a PR to do the refactor (after this one gets merged).

Reply:
TIL about type applications. Why would type applications make sense here? I've briefly browsed through the proposal and the way I understood it is to be more explicit where ambiguity exists. Does the ambiguity here come with the use of `show`?

Reply:
@endgame ah, nevermind, I understand what you were suggesting now.  Besides, there's no way to get rid of the proxy argument, otherwise GHC will not accept the class definition :)  But... it can wait for another PR.

Reply:
@romanofski it's nothing to do with `show`.  Just makes constructing `Proxy` values a bit cleaner, among other uses.
```
λ> :set -XTypeApplications
λ> import Data.Proxy
λ> :t Proxy @Int
Proxy @Int :: Proxy Int
```